internal/factsengine/gatherers/corosynccmapctl.go

```diff
@@ -45,6 +44,38 @@ func NewCorosyncCmapctlGatherer(executor utils.CommandExecutor) *CorosyncCmapctl
 	}
 }
 
+func corosyncCmapctlOutputToMap(corosyncCmapctlOutput string) *entities.FactValueMap {
+	outputMap := &entities.FactValueMap{Value: make(map[string]entities.FactValue)}
+	var cursor *entities.FactValueMap
+
+	for _, line := range strings.Split(corosyncCmapctlOutput, "\n") {
+		cursor = outputMap
+
+		value := strings.Split(line, "= ")[1]
+
+		pathAsString := strings.Split(line, " (")[0]
+		path := strings.Split(pathAsString, ".")
+
+		for i, key := range path {
+			currentMap := cursor
+
+			if i == len(path)-1 {
+				currentMap.Value[key] = entities.ParseStringToFactValue(value)
+
+				break
+			}
+
+			if _, found := currentMap.Value[key]; !found {
```

Comment:
I see a problem or at least a differentiation with other gatherers. Lists are not used.

When we have this:
```
nodelist.node.0.nodeid (u32) = 1
nodelist.node.0.ring0_addr (str) = 10.80.1.11
nodelist.node.1.nodeid (u32) = 2
nodelist.node.1.ring0_addr (str) = 10.80.1.12
```
We should treat the `0` and `1` values as list indexes instead of map keys.
Otherwise, when we want to loop them on rhai, we would need to loop them as maps, where the key is the index, which is strange.

I know that this makes the code a bit more difficult, but I think is the correct choice.
What do you think?

Reply:
We had a conversation with @fabriziosestito on Slack regarding this. I think you are right, a list with be ideal here but it increases the complexity of the code by quite a lot AFAICS. This way is just simpler, less code to maintain and just works.

Reply:
Also, it will make it harder to document that when the config shows `0` and `1` you need to expect an array.
The output is presented as a nested map and this way what we return is always consistent.

Reply:
I'm not sure on this thinking...
I would expect a list honestly, as the are actually lists, where the number is the index.
Anyway, I already had talked with Jaime on this, and due some few inconsystencies in the output, we already agreed that using the map is the unique viable option.
This will force as to use `values()` in these entries in `rhai`, but well it is what it is

Reply:
And mostly, the major inconsistency comes comparing in it with the other `corosync-conf` gatherer, which actually returns lists for the same values. So let's document this properly

Reply:
I totally see your point, as I am working on the documentation of the other gatherers and I realized that I was wrong, and the inconsistency will be in this gatherer. 
I am having second thoughts, so I am ok if you guys decide to go with the array solution.

>  due some few inconsistencies in the output we already agreed that using the map is the unique viable option

What are these inconsistencies about?

Reply:
Outputs like this:
```
runtime.services.cmap.0.rx (u64) = 3
runtime.services.cmap.0.tx (u64) = 2
runtime.services.cmap.service_id (u16) = 0
```
Which at the end mix what it should be a list with other fields...

Reply:
@jamie-suse by the way, could we add other test for this particular scenario?